posts/2021-09-24-travis-security.rst

```diff
@@ -0,0 +1,46 @@
+.. post:: 2021-09-24
+   :tags: security
+   :author: mrb
+
+.. role:: raw-html(raw)
+   :format: html
+
+
+Travis CI Security Incident
+===========================
+
+On September 9, 2021 one of our core devs discovered that artifacts building on
+Travis CI were being uploaded to our conda channel from PRs running on forked
+repositories. A quick investigation revealed that Travis CI was passing encrypted
+secrets to PR builds on forks. Further examination of our logs and artifacts indicated
+that this had been happening since about September 3, 2021. See this `CVE <https://nvd.nist.gov/vuln/detail/CVE-2021-41077>`_
+for more details on this incident. **As far as we know, there were no actual exploits against
+conda-forge which used this vulnerability.**
+
+Our Response
+------------
+
+We took the following steps to respond to this incident.
+
+1. We immediately turned off all builds on Travis CI by suspending the Travis CI GitHub App.
+2. We immediately disclosed the bug to Travis CI through our contacts there.
+3. Once Travis CI indicated to us that they were ready, we rotated
+   all feedstock tokens and later our anaconda.org token for our staging channel. The anaconda.org token
+   for the main ``conda-forge`` channel was never disclosed in this incident. Further, only ~70 feedstocks
+   had their tokens exposed in this incident.
+4. We examined our artifacts and marked as broken any artifacts that were uploaded from PRs. We think we found 
+   everything, but we are not completely sure.
```

Comment:
Is it correct to say that we cast a "wide net" and marked as broken more than we should to be in the safe side? I have to confess I got lost in the conversation about this.

Reply:
yes